lib/puppet/util/tagging.rb

```diff
@@ -78,16 +78,14 @@ def set_tags(tag_source)
     @tags = tag_source.tags
   end
 
-  # Return a copy of the tag list, so someone can't ask for our tags
-  # and then modify them.
   def tags
     @tags ||= new_tags
-    @tags.dup
+    @tags
```

Comment:
definitely need a second opinion on the safety of this change

Reply:
This makes me a little nervous, but acceptance tests pass. What do you think?